src/main/java/Snippet.java

```diff
@@ -0,0 +1,92 @@
+import java.util.Arrays;
+import java.util.Date;
+
+public class Snippet {
+    private String pathToJson;
+    private String title;
+    private String content;
+    private String language;
+    private String[] tags;
+    private Date created;
+    private Date modified;
+
+    Snippet(String pathToJson, String title, String content, String language, String[] tags) {
+        this.pathToJson = pathToJson;
+        this.title = title;
+        this.content = content;
+        this.language = language;
+        this.tags = tags;
+        this.created = new Date();
+        this.modified = new Date();
+
+        writeSnippetToJson();
+    }
+
+    void edit() {
+        new Editor(this);
+    }
+
+    private void writeSnippetToJson() {
+        JsonIO.getInstance().writeToJson(pathToJson, this);
+    }
+
+    private void onModification() {
+        this.modified = new Date();
+        writeSnippetToJson();
+    }
+
+    String getTitle() {
+        return title;
```

Comment:
might be cleaner to use this.title here since you use this.title in setTitle as well
same for the other get functions